salalib/alllinemap.cpp

```diff
@@ -173,55 +174,71 @@ std::tuple<std::unique_ptr<ShapeGraph>, std::unique_ptr<ShapeGraph>> AllLineMap:
    std::map<RadialKey,RadialSegment> radialsegs;
 
    // now make radial segments from the radial lines... (note, start at 1)
-   for (i = 1; i < m_radial_lines.size(); i++) {
-      if (m_radial_lines[i].vertex == m_radial_lines[i-1].vertex) {
-         if (m_radial_lines[i].ang == m_radial_lines[i-1].ang) {
+   auto iter = m_radial_lines.begin();
+   auto prevIter = m_radial_lines.begin();
+   ++iter;
+   for (;iter != m_radial_lines.end();) {
+      if (iter->vertex == prevIter->vertex) {
+         if (iter->ang == prevIter->ang) {
+            ++iter;
+            ++prevIter;
             continue;
          }
          else {
-            radialsegs.insert(std::make_pair( (RadialKey)m_radial_lines[i],
-                                              (RadialSegment)m_radial_lines[i-1]));
+            radialsegs.insert(std::make_pair( (RadialKey)(*iter),
+                                              (RadialSegment)(*prevIter)));
          }
       }
+      ++iter;
+      ++prevIter;
    }
 
    // and segment divisors from the axial lines...
    // TODO: (CS) Restructure this to get rid of all those brittle parallel data structure
    auto axIter = ax_radial_cuts.begin();
    auto axSeg = ax_seg_cuts.begin();
    for (i = 0; i < getAllShapes().size(); i++) {
-      for (size_t j = 1; j < axIter->second.size(); j++) {
+      auto axRadCutIter = axIter->second.begin();
+      auto axRadCutIterPrev = axIter->second.begin();
+      ++axRadCutIter;
+      for (size_t j = 1; j < axIter->second.size(); ++j) {
          // note similarity to loop above
-         RadialKey rk_end = m_radial_lines[axIter->second[j]];
-         RadialKey rk_start = m_radial_lines[axIter->second[j-1]];
+         RadialKey& rk_end = m_radial_lines[size_t(*axRadCutIter)];
+         RadialKey& rk_start = m_radial_lines[size_t(*axRadCutIterPrev)];
          if (rk_start.vertex == rk_end.vertex) {
             auto radialSegIter = radialsegs.find(rk_end);
             if (radialSegIter != radialsegs.end() && rk_start == radialSegIter->second.radial_b) {
                radialSegIter->second.add(axIter->first);
-               axSeg->second.add(std::distance(radialsegs.begin(), radialSegIter));
+               axSeg->second.insert(std::distance(radialsegs.begin(), radialSegIter));
             }
          }
+         ++axRadCutIter;
+         ++axRadCutIterPrev;
       }
       axIter++;
       axSeg++;
    }
 
    // and a little more setting up: key vertex relationships
-   prefvec<pvecint> keyvertexconns;
+   std::vector<std::vector<int> > keyvertexconns;
    int *keyvertexcounts = new int [m_keyvertexcount];
    for (int x = 0; x < m_keyvertexcount; x++) {
       keyvertexcounts[x] = 0;
    }
    // this sets up a two step relationship: looks for the key vertices for all lines connected to you
    for (size_t y = 0; y < m_connectors.size(); y++) {
-      keyvertexconns.push_back(pvecint());
+      keyvertexconns.push_back(std::vector<int>());
```

Comment:
Could do `auto &conn = keyvertexconns.back();` here instead of doing this inside loop further down.